### PR TITLE
mgr/dashboard: provide ability to edit cephfs subvolume 

### PR DESCRIPTION
**Pull request by nizamial09**
Currently only resizing is possible. Because once the item is created with the given values, then the metadata cannot be changed later on!

https://github.com/ceph/ceph/assets/71764184/b86c8b16-7aca-40b8-9ff4-4c358591d569

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/cephfs.py [hunk 1]
```diff
@@ -674,3 +674,14 @@ def create(self, vol_name: str, subvol_name: str, **kwargs):
             )
 
         return f'Subvolume {subvol_name} created successfully'
+
+    def set(self, vol_name: str, subvol_name: str, size: str):
+        if size:
```
Comment (pereman2) [inline src/pybind/mgr/dashboard/controllers/cephfs.py:679]: size is a string?

**Review by pereman2 — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]
LGTM
[Screencast from 2023-08-08 22-18-56.webm](https://github.com/ceph/ceph/assets/6695156/f0d7179b-860f-4e01-bc5f-d41787947cf1)

File: src/pybind/mgr/dashboard/controllers/cephfs.py (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/cephfs.py:679]: they have `infinite` as a size value when quota is not set.

**Comment by nizamial09** [comment]
goes after https://github.com/ceph/ceph/pull/52645

**Comment by cloudbehl** [comment]
@avanthakkar Can you add this fix as part of your PR.

```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-group/cephfs-subvolume-group.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-group/cephfs-subvolume-group.component.ts
index e1fc307afa..c67148d1a5 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-group/cephfs-subvolume-group.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-subvolume-group/cephfs-subvolume-group.component.ts
@@ -1,4 +1,4 @@
-import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import { Component, Input, OnChanges, OnInit, ViewChild } from '@angular/core';
 import { Observable, ReplaySubject, of } from 'rxjs';
 import { catchError, shareReplay, switchMap } from 'rxjs/operators';
 
@@ -21,7 +21,7 @@ import { Permissions } from '~/app/shared/models/permissions';
   templateUrl: './cephfs-subvolume-group.component.html',
   styleUrls: ['./cephfs-subvolume-group.component.scss']
 })
-export class CephfsSubvolumeGroupComponent implements OnInit {
+export class CephfsSubvolumeGroupComponent implements OnInit, OnChanges {
   @ViewChild('quotaUsageTpl', { static: true })
   quotaUsageTpl: any;
```

**Comment by cloudbehl** [comment]
jenkins test make check

**Comment by avanthakkar** [comment]
jenkins test dashboard cephadm

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Review by avanthakkar — APPROVED** [review]
lgtm!

**Pull request merged**